web/src/main/java/org/springframework/security/web/ObservationFilterChainDecorator.java

```diff
@@ -568,13 +569,12 @@ public String getContextualName(FilterChainObservationContext context) {
 
 		@Override
 		public KeyValues getLowCardinalityKeyValues(FilterChainObservationContext context) {
-			KeyValues kv = KeyValues.of(CHAIN_SIZE_NAME, String.valueOf(context.getChainSize()))
+			return KeyValues.of(CHAIN_SIZE_NAME, String.valueOf(context.getChainSize()))
 					.and(CHAIN_POSITION_NAME, String.valueOf(context.getChainPosition()))
-					.and(FILTER_SECTION_NAME, context.getFilterSection());
-			if (context.getFilterName() != null) {
-				kv = kv.and(FILTER_NAME, context.getFilterName());
-			}
-			return kv;
+					.and(FILTER_SECTION_NAME, context.getFilterSection())
+					.and(FILTER_NAME, (context.getFilterName() != null)
+							? context.getFilterName()
+							: KeyValue.NONE_VALUE);
```

Comment:
Are you able to add a unit test in `ObservationFilterChainDecoratorTests` and `ObservationWebFilterChainDecoratorTests` that confirms that the observation handler gets `NONE_VALUE` for the value in this case?

Reply:
Sure, I will try my best.
Before I start from scratch, is there already a test that verifies tagging? I couldn't find one. Thanks!

Reply:
Here's a sample to get you started:

```java
@Test
void decorateFiltersWhenCompletesThenHasSpringSecurityReachedFilterNameTag() throws Exception {
	ObservationHandler<Observation.Context> handler = mock(ObservationHandler.class);
	given(handler.supportsContext(any())).willReturn(true);
	ObservationRegistry registry = ObservationRegistry.create();
	registry.observationConfig().observationHandler(handler);
	ObservationFilterChainDecorator decorator = new ObservationFilterChainDecorator(registry);
	FilterChain chain = mock(FilterChain.class);
	Filter filter = spy(new BasicAuthenticationFilter());
	FilterChain decorated = decorator.decorate(chain, List.of(filter));
	decorated.doFilter(new MockHttpServletRequest("GET", "/"), new MockHttpServletResponse());
	ArgumentCaptor<Observation.Context> context = ArgumentCaptor.forClass(Observation.Context.class);
	verify(handler, times(3)).onScopeClosed(context.capture());
	assertThat(context.getValue().getLowCardinalityKeyValue("spring.security.reached.filter.name"))
		.isEqualTo("BasicAuthenticationFilter");
}
```

Let me know if you get stuck. I think this would need to change to simulate the circumstance where the filter name is null.